### PR TITLE
Improvements for Android TV

### PR DESCRIPTION
**Pull request by lihanchen**
To build for Android TV, user need to modify the orientation from "portrait" to "landscape"
for the four activities in tensorflow/examples/android/AndroidManifest.xml

1. Show a toast when no camera is detected
2. Game Controller L1 button and remote controller DRAP_Center also triggers debug output
3. Fix the layout of Stylize app on Android TV. The style selector will be on right side
4. User can use DPAD to navigate through styles in Stylize app

### CODE REVIEW

File: tensorflow/contrib/makefile/README.md [hunk 1]
```diff
@@ -194,6 +194,8 @@ with:
 srcs = glob(["libs/arm64-v8a/*.so"]),
 ```
 
+If you are build for Android TV (Shield TV devices), replace "portrait" with "landscape" for android:screenOrientation in four activities in tensorflow/examples/android/AndroidManifest.xml
```
Comment (andrewharp) [inline tensorflow/contrib/makefile/README.md]: If you are building for Android TV (Shield TV devices), replace "portrait" with "landscape" for android:screenOrientation in all four activities in tensorflow/examples/android/AndroidManifest.xml

File: tensorflow/examples/android/src/org/tensorflow/demo/StylizeActivity.java [hunk 2]
```diff
@@ -602,4 +622,36 @@ private void renderDebug(final Canvas canvas) {
...
+    float highestValue = 0;
+    for (int i = 0; i < adapter.getCount(); i++) {
+      if (adapter.items[i].value > highestValue) {
+        currentSelect = i;
+        highestValue = adapter.items[i].value;
+      }
+    }
+    setStyle(adapter.items[(currentSelect + moveOffset + adapter.getCount()) % adapter.getCount()], 1);
+
+    return super.onKeyDown(keyCode, event);
```
Comment (andrewharp) [inline tensorflow/examples/android/src/org/tensorflow/demo/StylizeActivity.java]: Do you need to return the key event if it's already been handled by this method?

File: tensorflow/examples/android/src/org/tensorflow/demo/StylizeActivity.java [hunk 3]
```diff
@@ -55,6 +59,11 @@
 import org.tensorflow.demo.env.ImageUtils;
 import org.tensorflow.demo.env.Logger;
 import org.tensorflow.demo.R; // Explicit import needed for internal Google builds.
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Vector;
```
Comment (andrewharp) [inline tensorflow/examples/android/src/org/tensorflow/demo/StylizeActivity.java]: These imports should go at the top.

File: tensorflow/examples/android/src/org/tensorflow/demo/StylizeActivity.java [hunk 4]
```diff
@@ -381,6 +390,17 @@ public void drawCallback(final Canvas canvas) {
     grid = (GridView) findViewById(R.id.grid_layout);
     grid.setAdapter(adapter);
     grid.setOnTouchListener(gridTouchAdapter);
+
+    // Change UI on Android TV
+    UiModeManager uiModeManager = (UiModeManager) getSystemService(UI_MODE_SERVICE);
+    if (uiModeManager.getCurrentModeType() == Configuration.UI_MODE_TYPE_TELEVISION) {
+      RelativeLayout.LayoutParams layoutParams = new RelativeLayout.LayoutParams(400, ViewGroup.LayoutParams.MATCH_PARENT);
```
Comment (andrewharp) [inline tensorflow/examples/android/src/org/tensorflow/demo/StylizeActivity.java]: Where does the 400 come from? Can it be computed programmatically?

File: tensorflow/contrib/makefile/README.md (hunk 1 above)
Comment (lihanchen) [reply tensorflow/contrib/makefile/README.md]: Fixed

File: tensorflow/examples/android/src/org/tensorflow/demo/StylizeActivity.java (hunk 2 above)
Comment (lihanchen) [reply tensorflow/examples/android/src/org/tensorflow/demo/StylizeActivity.java]: Changed behavior. Only pass the key down event if it is not DPAD key.

File: tensorflow/examples/android/src/org/tensorflow/demo/StylizeActivity.java (hunk 3 above)
Comment (lihanchen) [reply tensorflow/examples/android/src/org/tensorflow/demo/StylizeActivity.java]: These are not needed, removed.

File: tensorflow/examples/android/src/org/tensorflow/demo/StylizeActivity.java (hunk 4 above)
Comment (lihanchen) [reply tensorflow/examples/android/src/org/tensorflow/demo/StylizeActivity.java]: I am not sure how the textureView adjust its width in runtime.
Can you point me where I can read the width?

File: tensorflow/examples/android/src/org/tensorflow/demo/StylizeActivity.java [hunk 5]
```diff
@@ -43,18 +47,20 @@
...
-import java.util.Collections;
-import java.util.Vector;
+
 import org.tensorflow.contrib.android.TensorFlowInferenceInterface;
 import org.tensorflow.demo.OverlayView.DrawCallback;
 import org.tensorflow.demo.env.BorderedText;
 import org.tensorflow.demo.env.ImageUtils;
 import org.tensorflow.demo.env.Logger;
-import org.tensorflow.demo.R; // Explicit import needed for internal Google builds.
+
```
Comment (andrewharp) [inline tensorflow/examples/android/src/org/tensorflow/demo/StylizeActivity.java]: extra whitespace

File: tensorflow/examples/android/src/org/tensorflow/demo/StylizeActivity.java (hunk 3 above)
Comment (andrewharp) [reply tensorflow/examples/android/src/org/tensorflow/demo/StylizeActivity.java]: Still seem to be here?

**Review by andrewharp — APPROVED** [review]

**Pull request merged**